lib/grant-types/client-credentials-grant-type.js

```diff
@@ -0,0 +1,112 @@
+
+/**
+ * Module dependencies.
+ */
+
+var AbstractGrantType = require('./abstract-grant-type');
+var InvalidArgumentError = require('../errors/invalid-argument-error');
+var InvalidGrantError = require('../errors/invalid-grant-error');
+var Promise = require('bluebird');
+var util = require('util');
+
+/**
+ * Constructor.
+ */
+
+function ClientCredentialsType(options) {
+  options = options || {};
+
+  if (!options.model) {
+    throw new InvalidArgumentError('Missing parameter: `model`');
+  }
+
+  if (!options.model.getUserFromClient) {
+    throw new InvalidArgumentError('Invalid argument: model does not implement `getUserFromClient()`');
+  }
+
+  if (!options.model.saveToken) {
+    throw new InvalidArgumentError('Invalid argument: model does not implement `saveToken()`');
+  }
+
+  AbstractGrantType.call(this, options);
+}
+
+/**
+ * Inherit prototype.
+ */
+
+util.inherits(ClientCredentialsType, AbstractGrantType);
+
+/**
+ * Handle client credentials grant.
+ *
+ * @see https://tools.ietf.org/html/rfc6749#section-4.4.2
+ */
+
+ClientCredentialsType.prototype.handle = function(request, client) {
```

Comment:
Same note as above - something like `validateAndAuthorize` or `processGrant`.  `handle` is just too vague.